csrc/scheduler/nvmmh.cpp

```diff
@@ -0,0 +1,259 @@
+    nvmmh_func::func =                                                 \
+        reinterpret_cast<decltype(&func)>(dlsym(nvmmh_handle, #func)); \
+  }
+
+  // TODO: check that version of lib matches version we compiled against
+  DEFINE_NVMMH_SYMBOL(nvMatmulHeuristicsGetVersionMajor)
+  DEFINE_NVMMH_SYMBOL(nvMatmulHeuristicsGetVersionMinor)
+  unsigned lib_major = nvmmh_func::nvMatmulHeuristicsGetVersionMajor();
+  unsigned lib_minor = nvmmh_func::nvMatmulHeuristicsGetVersionMinor();
+  if (lib_major != NVMMH_VERSION_MAJOR || lib_minor != NVMMH_VERSION_MINOR) {
+    TORCH_WARN_ONCE(
+        "nvFuser was compiled against nvMatmulHeuristics version ",
+        NVMMH_VERSION_MAJOR,
+        ".",
+        NVMMH_VERSION_MINOR,
+        " but found nvMatmulHeuristics shared library version ",
+        lib_major,
+        ".",
+        lib_minor,
+        ". Exactly matching versions are required");
+  }
+
+  ALL_NVMMH_API_WRAPPER(DEFINE_NVMMH_SYMBOL);
+
+#undef DEFINE_NVMMH_SYMBOL
+
+  return true;
+}
+
+#undef ALL_NVMMH_API_WRAPPER
+
+int isValidScaledGemmConfig(const nvmmhKernelConfiguration_t* result) {
+  std::array<uint16_t, 3> Cta;
+  Cta[0] = result->cta[0];
+  Cta[1] = result->cta[1];
+  Cta[2] = result->cta[2];
+
+  // https://github.com/NVIDIA/cutlass/blob/c6aeb9179c5f74a0fcdbd28527bf4b6ba8c60752/include/cutlass/gemm/collective/builders/sm100_common.inl#L693-L701
+  if (Cta[0] != 128) {
+    return false;
+  }
+
+  if (Cta[1] != 64 && Cta[1] != 128 && Cta[1] != 192 && Cta[1] != 256) {
+    return false;
+  }
+
+  return true;
+}
+
+#else // HAS_NVMMH
+
+bool initNVMMH() {
+  TORCH_WARN_ONCE("nvFuser was built without nvMatmulHeuristics support");
+  return false;
+}
+
+#define NVMMH_SAFE_CALL(x)         \
+  NVF_THROW("Attempted to call " x \
+            " but nvFuser was not built with nvMatmulHeuristics support");
+
+#endif // HAS_NVMMH
+
+GemmTile getProblemSize(Fusion* fusion, SchedulerRuntimeInfo& runtime_info) {
+  const std::vector<ScaledMmaOp*> scaled_mmas =
+      ir_utils::getOpsOfType<ScaledMmaOp>(fusion);
+
+  NVF_ERROR_EQ(
+      scaled_mmas.size(),
+      1,
+      "Cutlass scheduler expects exactly one ScaledMmaOp");
+
+  ScaledMmaOp* mma = scaled_mmas.front();
+
+  NVF_ERROR(mma->matrix1()->isFusionInput());
+  NVF_ERROR(mma->matrix2()->isFusionInput());
+  const std::vector<int64_t>& a_shape =
+      runtime_info.getInputAllocationSizes(mma->matrix1());
+  const std::vector<int64_t>& b_shape =
+      runtime_info.getInputAllocationSizes(mma->matrix2());
+
+  NVF_ERROR_EQ(a_shape.size(), 2);
+  NVF_ERROR_EQ(b_shape.size(), 2);
+  int64_t m = a_shape.front();
+  int64_t n = b_shape.back();
+  int64_t k = a_shape.back();
+  return {m, n, k};
+}
+
+} // namespace
+
+void fillNvMatmulHeuristicsParams(
+    CutlassParams* params,
+    Fusion* fusion,
+    SchedulerRuntimeInfo& runtime_info) {
+  FUSER_PERF_SCOPE("fillNvMatmulHeuristicsParams");
+  if (initNVMMH()) {
+#ifdef HAS_NVMMH
+    nvmmhHandle_t handle = nullptr;
```

Comment:
More of a nitpick. I think a single `HAS_NVMMH` macro block is more readable. This if-statement block can be moved to a helper function in the anonymous namespace.